docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-  to the production cluster. In addition, there are long-term differences between the cluster resources (e.g. different
-  security keys, different database endpoints etc ...).
-* Federation. Different clusters run in separate regions, usually with very similar resources but different
-  configurations.
+For each path supplied with the flag `--git-path` (or `.` if none are
+supplied),
 
-The main goal of `.flux.yaml` configuration files is to help deploying similar resources/clusters:
+ - if a `.flux.yaml` file is found, it is used **instead** of looking
+   for YAML files;
 
-* with minimal replication of resource definitions
-* while keeping Flux neutral about the factorization technology used
+ - if no `.flux.yaml` file is found, the usual behaviour of looking
+   for YAML files is used.
 
-## File-access behaviour in Flux
+The manifests from all the `--git-path` values -- read from YAML files
+or generated -- are combined before applying to the cluster. If
+duplicates are detected, an error is logged and the sync is abandoned.
 
-Flux performs two types of actions on raw manifest files from the Git repository:
+A `.flux.yaml` file can be _at_ the path mentioned, or in a directory
+_above_ it in the git repo. Here are some examples:
 
-1. Read manifest files when performing a sync operation (i.e making sure that the status of the cluster reflects what's
-   in the manifest files, adjusting it if necessary)
-2. Update the manifest files of [workload](fluxctl.html#workloads).
-   Specifically, flux can update:
-    * container images, when releasing a new image version. A release can happen manually or automatically, when a new
-      container image is pushed to a repository.
-    * annotations, which establish the release policy of a workload (e.g. whether it should be automatically released,
-      whether it should be locked from releasing, what image tags should be considered for automated releases …)
+```sh
+# show normal files _and_ .flux.yaml
+flux-kustomization-example$ tree -a -P .flux.yaml -P '*' -I .git
+.
+├── base
+│   ├── demo-ns.yaml
+│   ├── kustomization.yaml
+│   ├── podinfo-dep.yaml
+│   ├── podinfo-hpa.yaml
+│   └── podinfo-svc.yaml
+├── .flux.yaml
+├── LICENSE
+├── production
+│   ├── flux-patch.yaml
+│   ├── kustomization.yaml
+│   └── replicas-patch.yaml
+├── README.md
+└── staging
+    ├── flux-patch.yaml
+    └── kustomization.yaml
+```
 
-Flux can be configured to confine the scope of (1) and (2):
+In this case, say you started `fluxd` with `--git-path=staging`, it
+would find `.flux.yaml` in the top directory and use that to generate
+manifests.  The files and directories (if there were any) in
+`staging/` are ignored, in favour of the instructions given in the
+`.flux.yaml` file.
 
-* To specific (sub)directories (flag `--git-path`)
-* To a Git branch other than `master` (flag `--git-branch`)
+In this modified example,
```

Comment:
I think it's worth indicating what is being changed (I had to look at it a couple of times).

Reply:
Yes good point that. Also adopted your suggestion below of sticking with `staging` in the example.